semantics/executable-spec/src/Data/Pulse.hs

```diff
@@ -43,20 +53,25 @@ data PulseListM m ans where
 
 instance Show ans => Show (PulseListM m ans) where
   show(PulseList ass n _ t a) = "(Pulse "++assoc ass++show n++status t++show a++")"
-    where status [] = " Done "
-          status (_ : _) = " More "
-          assoc LeftA = "left "
+    where assoc LeftA = "left "
           assoc RightA = "right "
 
+isNil :: [a] -> Bool
```

Comment:
Why not use `null`? https://hackage.haskell.org/package/base-4.14.1.0/docs/Prelude.html#v:null